### PR TITLE
Soumya

### PR DESCRIPTION
**Pull request by hellosibun**
added item entity dependency

### CODE REVIEW

File: FoodOrderingApp-api/src/main/java/com/upgrad/FoodOrderingApp/api/controller/CategoryController.java [hunk 1]
```diff
@@ -1,94 +1,95 @@
...
+import java.util.UUID;
+
+// Category Controller Handles all  the Category related endpoints
+
+@RestController
+@RequestMapping("/category")
+public class CategoryController {
+
+    @Autowired
+    CategoryService categoryService;
```
Comment (ashwini008) [inline FoodOrderingApp-api/src/main/java/com/upgrad/FoodOrderingApp/api/controller/CategoryController.java]: Shouldn't this be private?

Comment (hellosibun) [reply FoodOrderingApp-api/src/main/java/com/upgrad/FoodOrderingApp/api/controller/CategoryController.java]: no

File: FoodOrderingApp-api/src/main/java/com/upgrad/FoodOrderingApp/api/controller/OrderController.java [hunk 2]
```diff
@@ -1,25 +1,240 @@
...
-//        //get access-token from the request header
-//        String accessToken = authorization.split("Bearer ")[1];
-//
-//        //Calls getCouponByCouponName of orderService to get the coupon by name from DB
-////        CouponEntity couponEntity;
-//
-//    }
+
+    @Autowired
+    OrderService orderService; // Handles all the Service Related Order.
```
Comment (ashwini008) [inline FoodOrderingApp-api/src/main/java/com/upgrad/FoodOrderingApp/api/controller/OrderController.java]: private for all @Autowired declaration. Refer all files in this PR

File: FoodOrderingApp-service/src/main/java/com/upgrad/FoodOrderingApp/service/businness/ItemService.java [hunk 3]
```diff
@@ -1,20 +1,78 @@
...
+
+    @Autowired
+    private OrderItemDao orderItemDao;
+
+    @Autowired
+    private UtilityProvider utilityProvider;
+
+    /* This method is to get Items By Popularity and returns list of ItemEntity it takes restaurantEntity as input.
+     */
+    public List<ItemEntity> getItemsByPopularity(RestaurantEntity restaurantEntity) {
```
Comment (ashwini008) [inline FoodOrderingApp-service/src/main/java/com/upgrad/FoodOrderingApp/service/businness/ItemService.java]: Any reason for overriding the implementation I wrote?

File: FoodOrderingApp-api/src/main/java/com/upgrad/FoodOrderingApp/api/controller/OrderController.java (hunk 2 above)
Comment (hellosibun) [reply FoodOrderingApp-api/src/main/java/com/upgrad/FoodOrderingApp/api/controller/OrderController.java]: is it really necessary to make private???

File: FoodOrderingApp-service/src/main/java/com/upgrad/FoodOrderingApp/service/businness/ItemService.java (hunk 3 above)
Comment (hellosibun) [reply FoodOrderingApp-service/src/main/java/com/upgrad/FoodOrderingApp/service/businness/ItemService.java]: no reason even if u make it private still output came the same

**Pull request merged**